tests/pages/firefox/os/devices.py

```diff
@@ -0,0 +1,98 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait as Wait
+from selenium.webdriver.support.select import Select
+
+from ..base import FirefoxBasePage
+from ...page import PageRegion
+from ...regions.modal import Modal
+
+
+class DevicesPage(FirefoxBasePage):
+
+    _url = '{base_url}/{locale}/firefox/os/devices'
+
+    _location_select_locator = (By.ID, 'location')
+    _get_phone_button_locator = (By.CSS_SELECTOR, '.location-select-wrapper button')
+    _phone_thumbnail_locator = (By.CSS_SELECTOR, '#smartphones li:first-child > .device-thumbnail')
+    _phone_detail_locator = (By.CSS_SELECTOR, '.smartphone-detail-list .device-detail:first-child')
+    _tv_thumbnail_locator = (By.CSS_SELECTOR, '#tvs li:first-child > .device-thumbnail')
+    _tv_detail_locator = (By.CSS_SELECTOR, '.tvs-detail-list .device-detail:first-child')
+
+    def select_location(self, value):
+        assert self.is_button_disabled, 'Get phone button is already enabled'
+        el = self.selenium.find_element(*self._location_select_locator)
+        Select(el).select_by_visible_text(value)
+        Wait(self.selenium, self.timeout).until(lambda s: not self.is_button_disabled)
+
+    @property
+    def is_button_disabled(self):
+        el = self.selenium.find_element(*self._get_phone_button_locator)
+        return el.get_attribute('disabled') == 'true'
+
+    def open_modal(self):
+        modal = Modal(self.selenium)
+        self.selenium.find_element(*self._get_phone_button_locator).click()
+        Wait(self.selenium, self.timeout).until(lambda s: modal.is_displayed)
+        return modal
+
+    def open_phone_detail(self):
+        el = self.selenium.find_element(*self._phone_detail_locator)
+        detail = DeviceDetail(self.selenium, root=el)
+        assert not detail.is_displayed, 'Phone detail is already displayed'
```

Comment:
I'm not sure we need this assert. It would prevent a test from opening another phone details when one is already open, which is a valid scenario.